### PR TITLE
Use the `KOTLIN_OFFICIAL` code style defaults for Kotlin

### PR DESCRIPTION
**Pull request by KushalP**
JetBrains both ships the language and an editor for Kotlin. We should conform to their house style where it makes sense to us.

### CODE REVIEW

**Review by holzensp — CHANGES_REQUESTED** [review]
We could do this, but it requires that `spotless` is setup to use the same style (see [here](https://github.com/apple/pkl/blob/main/buildSrc/src/main/kotlin/pklKotlinLibrary.gradle.kts#L24))

**Comment by KushalP** [comment]
Thanks. Updated that line to use `.kotlinlangStyle()`.

**Comment by KushalP** [comment]
I've separated the configuration changes from the `spotlessApply` changes.

**Comment by KushalP** [comment]
It will be easiest to review this PR with whitespace hidden:

Screenshot 2024-02-05 at 13 13 23

**Comment by KushalP** [comment]
And the original (IntelliJ) configuration change came from the following editor suggestion:

Screenshot 2024-02-05 at 12 38 21

**Comment by KushalP** [comment]
I'm going to close this for now as it's less important than other things. Should anyone come across this, it's possible to ignore the pop-up from IntelliJ and use the Pkl project's configuration.

File: pkl-codegen-java/src/main/kotlin/org/pkl/codegen/java/JavaCodeGenerator.kt [hunk 1]
```diff
@@ -40,894 +40,931 @@ import org.pkl.core.util.CodeGeneratorUtils
...
+                                propertyType = propertyType.baseType
+                            }
+                        }
+                        propertyType is PType.Class && propertyType.pClass == pClass
+                    }
+                if (modulePropertiesWithMatchingType.size == 1) {
+                    // exactly one module property has this type -> make it available for direct
+                    // injection
+                    // (potential improvement: make type available for direct injection if it occurs
+                    // exactly
```
Comment (holzensp) [inline pkl-codegen-java/src/main/kotlin/org/pkl/codegen/java/JavaCodeGenerator.kt:542]: That's... disappointing. I would have hoped spotless could line-wrap better [throughout]

**Review by holzensp — COMMENTED** [review]
Leaving it open for other maintainers to chime in.

I don't mind it and here-and-there prefer the Kotlin Language layout. That said, it's a high volume PR. I understand the inclination to default into the suggestion from the IntelliJ pop-up, but it might be too much churn for what it delivers (I cancel those pop-ups by default).

The vast majority of changes in this PR seem to be cause by having a narrower line-width. If that's tweakable, friction might drop significantly.

**Pull request closed without merging**